ref_arch/multi_tenant.rst

```diff
@@ -0,0 +1,393 @@
+  +  campaign_id bigint,
+     name text NOT NULL,
+     image_url text,
+     target_url text,
+     impressions_count bigint DEFAULT 0,
+     clicks_count bigint DEFAULT 0,
+     created_at timestamp without time zone NOT NULL,
+  -  updated_at timestamp without time zone NOT NULL
+  +  updated_at timestamp without time zone NOT NULL,
+  +  PRIMARY KEY (company_id, id),
+  +  FOREIGN KEY (company_id, campaign_id)
+  +    REFERENCES ads (company_id, id)
+   );
+
+   CREATE TABLE clicks (
+  -  id bigserial PRIMARY KEY,
+  -  ad_id bigint REFERENCES ads (id),
+  +  id bigserial,
+  +  company_id bigint,
+  +  ad_id bigint,
+     clicked_at timestamp without time zone NOT NULL,
+     site_url text NOT NULL,
+     cost_per_click_usd numeric(20,10),
+     user_ip inet NOT NULL,
+  -  user_data jsonb NOT NULL
+  +  user_data jsonb NOT NULL,
+  +  PRIMARY KEY (company_id, id),
+  +  FOREIGN KEY (company_id, ad_id)
+  +    REFERENCES ads (company_id, id)
+   );
+
+   CREATE TABLE impressions (
+  -  id bigserial PRIMARY KEY,
+  -  ad_id bigint REFERENCES ads (id),
+  +  id bigserial,
+  +  company_id bigint,
+  +  ad_id bigint,
+     seen_at timestamp without time zone NOT NULL,
+     site_url text NOT NULL,
+     cost_per_impression_usd numeric(20,10),
+     user_ip inet NOT NULL,
+  -  user_data jsonb NOT NULL
+  +  user_data jsonb NOT NULL,
+  +  PRIMARY KEY (company_id, id),
+  +  FOREIGN KEY (company_id, ad_id)
+  +    REFERENCES ads (company_id, id)
+   );
+
+The final schema is available for `download <https://examples.citusdata.com/tutorial/schema.sql>`_.
+
+Distributing Tables, Ingesting Data
+-----------------------------------
+
+.. note::
+
+  This guide is designed so you can follow along in your own Citus database. Use one of these alternatives to spin up a database:
+
+  * Run Citus locally using :ref:`single_machine_docker`, or
+  * Provision a cluster using `Citus Cloud <https://console.citusdata.com/users/sign_up>`_
+
+  You'll run the SQL commands using psql:
+
+  * **Docker**: :code:`docker exec -it citus_master psql -U postgres`
+  * **Cloud**: :code:`psql "connection-string"` where the connection string for your formation is available in the Cloud Console.
+
+  In either case psql will be connected to the coordinator node for the cluster.
+
+At this point feel free to follow along in your own Citus cluster by downloading and executing the SQL to create the schema. Once the schema is ready, we can tell Citus to create shards on the workers. From the coordinator node, run:
+
+::
+
+  SELECT create_distributed_table('companies',   'id');
+  SELECT create_distributed_table('campaigns',   'company_id');
+  SELECT create_distributed_table('ads',         'company_id');
+  SELECT create_distributed_table('clicks',      'company_id');
+  SELECT create_distributed_table('impressions', 'company_id');
+
+This activates these tables for distributed storage and query execution. The next step is loading sample data into the cluster.
```

Comment:
Could we have a slightly more technical description of `create_distributed_table`? Above, we've explained what shards are and what worker nodes are. So, we could may be say something like below (with much better wording) :-

`create_distributed_table` informs Citus which tables are distributed i.e. all incoming queries to these tables will be planned for distributed execution. It also creates shards for these tables on the worker nodes so that incoming data can be routed to them.`